lesson3_videos.py

```diff
@@ -84,29 +104,60 @@ def main():
     for person in Person:
         person.show()
 
+    depts = [
+        ('A101', 'Flight Controls', 'Andrew'),
+        ('B245', 'Stability and Control', 'Peter'),
+        ('Q456', 'Advanced Research', 'Susan'),
+        ('1234', 'Dept of bad naming conventions', 'Steven'),
+        ('C12345', 'Dept of too long numbers', 'Peter')
+    ]
+
+    for dept in depts:
+        try:
+            with db.transaction():
+                new_dept = Department.create(
+                    dept_number=dept[0],
+                    dept_name=dept[1],
+                    dept_manager=dept[2], )
+                new_dept.save()
+
+        except Exception as e:
+            logger.info(f'Error creating department = {dept[0]}')
+            logger.info(e)
+            logger.info('Database protects adding department when a person does not exist for it')
+
+    for dept in Department:
+        dept.show()
+
     jobs = [
-        ('Analyst', '2017-02-01', '2019-07-31', 34.999, 'Andrew'),
-        ('Developer', '2017-02-01', '2019-07-31', 34.999, 'Fred'),
+        ('Analyst', '2017-02-01', '2019-07-31', 34.999, 'Andrew', 'Flight Controls'),
```

Comment:
The issue here is that the name of the department (for example, _Flight Controls_) is not the primary key, for the _Department_ table. For advanced research, you should be using _A101_.

Also, note that you'll get another foreign key error in the next line, both for using _Advanced Research_ instead of _Q456_ and because _Fred_ is not an employee (he might be up to something ;-)).